frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,353 @@
+      spec: {
+        accessModes: {},
+        resources: {
+          requests: {
+            storage: '',
+          },
+        },
+        selector: {
+          matchLabels: {},
+        },
+      },
+    };
+
+    this.state = {
+      resourceObj: resourceObj,
+      inProgress: false,
+      pvcName: '',
+      accessMode: '',
+      requestSizeValue: '',
+      requestSizeUnit: '',
+      defaultRequestSizeValue: '1',
+      defaultRequestSizeUnit: 'Gi',
+      disableForm: false,
+      showKeyValue: false,
+      nameValuePairs: [['', '']],
+      radios: [
+        {
+          value: 'ReadWriteOnce',
+          title: <span className="form-right-padding">Single User (RWO)</span>,
+        },
+        {
+          value: 'ReadWriteMany',
+          title: <span className="form-right-padding">Shared Access (RWX)</span>,
+        },
+        {
+          value: 'ReadOnlyMany',
+          title: <span className="form-right-padding">Read Only (ROX)</span>,
+        },
+      ],
+      dropdownUnits: {
+        ['Mi']: 'Mi',
+        ['Gi']: 'Gi',
+        ['Ti']: 'Ti',
+      },
+    };
+  }
+  componentDidMount() {
+    this.setState({
+      accessMode: 'ReadWriteOnce',
+      requestSizeValue: this.state.defaultRequestSizeValue,
+      requestSizeUnit: this.state.defaultRequestSizeUnit,
+    });
+  }
+  onError = err => {
+    this.setState({
+      error: err,
+      inProgress: false,
+    });
+  };
+
+  handleChange = event => {
+    // this handles pvcName, accessMode, size
+    const name = event.target.name;
+    const value = event.target.value;
+    this.setState({ [name]: value } as Pick<PvcState, keyof PvcState>);
+  };
+
+  handleNameValuePairs = event => {
+    this.setState({ nameValuePairs: event.nameValuePairs });
+  };
+
+  onShowLabelSelectors = () => {
+    this.setState({ showKeyValue: !this.state.showKeyValue });
+  };
+
+  onFormDisable = disable => {
+    this.setState({
+      disableForm: disable,
+    });
+  };
+
+  validateForm = () => {
+    const { pvcName, accessMode, requestSizeUnit, requestSizeValue } = this.state;
+    // Check that a valid PVC has been choosen
+    if (!pvcName) {
+      this.setState({ error: 'Persistent volume claim name must be set.' });
+      return false;
+    }
+    if (!accessMode) {
+      this.setState({ error: 'Access mode must be set.' });
+      return false;
+    }
+    if (!requestSizeValue) {
+      this.setState({ error: 'Size must be set.' });
+      return false;
+    }
+    if (!requestSizeUnit) {
+      this.setState({ error: 'Unit for size must be set.' });
+      return false;
+    }
+    return true;
+  };
+
+  nameValueArrayToObject = nvpArray => {
+    return _.reduce(nvpArray, (acc, nv) => _.assign(acc, { [nv[0]]: nv[1] }), {});
+  };
+  // Create PVC //
+  save = e => {
+    e.preventDefault();
+    if (!this.validateForm()) {
+      return;
+    }
+
+    const { kindObj, namespace } = this.props;
+    let resourceObj = { ...this.state.resourceObj };
```

Comment:
This needs to be a deep clone. (Although if you follow my other comment, this object wouldn't be kept in state at all.)